play-dl/YouTube/utils/request.ts

```diff
@@ -44,7 +46,8 @@ export async function request(url : string, options? : RequestOpts): Promise<str
 
 export async function request_stream(url : string, options? : RequestOpts): Promise<IncomingMessage>{
     return new Promise(async (resolve, reject) => {
-        let res = await https_getter(url, options)
+        let res = await https_getter(url, options).catch((err) => err as string)
+        if(typeof res === 'string') throw new Error(res)
```

Comment:
throw new Error(res) to return reject(new Error(res))
because of correct handling.

Reply:
Okay thanks for correcting me